src/Control/Session.php

```diff
@@ -319,36 +320,38 @@ public function start(HTTPRequest $request)
                 session_save_path($session_path);
             }
 
-            // If we want a secure cookie for HTTPS, use a seperate session name. This lets us have a
-            // seperate (less secure) session for non-HTTPS requests. Note that if this causes problems
+            // If we want a secure cookie for HTTPS, use a separate session name. This lets us have a
+            // separate (less secure) session for non-HTTPS requests
             // if headers_sent() is true then it's best to throw the resulting error rather than risk
             // a security hole.
             if ($secure) {
                 session_name($this->config()->get('cookie_name_secure'));
             }
 
-            session_start();
+            $sessionParameters = [
+                "cookie_path" => $path,
+                "cookie_domain" => $domain ?: "",
+                "cookie_secure" => $secure,
+                "cookie_httponly" => true
+            ];
+
+            if ($timeout) {
```

Comment:
this `if` block feels to be rather redundant. also, for consistency we may want to move `cookie_lifetime` parameter definition into the array with other params. E.g.
```php
$sessionParameters = [
  ...
  "cookie_lifetime" => $timeout ?: "0"
  ...
];
```

Reply:
Huh, this got lost in the merge. I'll chuck that back in :)